scvi/utils/_differential.py

```diff
@@ -403,6 +422,70 @@ def scale_sampler(
         return dict(scale=px_scales, batch=batch_ids)
 
 
+def estimate_delta(lfc_means: List[np.ndarray], coef=0.6, min_thres=0.3):
+    """Computes a threshold LFC value based on means of LFCs.
+
+    Parameters
+    ----------
+    lfc_means
+        LFC means for each gene, should be 1d.
+    coef
+        Tunable hyperparameter to choose the threshold based on estimated modes, defaults to 0.6
+    min_thres
+        Minimum returned threshold value, defaults to 0.3
+    """
+    logger.debug("Estimating delta from effect size samples")
+    assert lfc_means.ndim == 1
+    gmm = GaussianMixture(n_components=3)
+    gmm.fit(lfc_means[:, None])
+    vals = np.sort(gmm.means_.squeeze())
+    res = coef * np.abs(vals[[0, -1]]).mean()
+    res = np.maximum(min_thres, res)
+    return res
+
+
+def estimate_pseudocounts_offset(
+    scales_a: List[np.ndarray],
+    scales_b: List[np.ndarray],
+    where_zero_a: List[np.ndarray],
+    where_zero_b: List[np.ndarray],
+):
+    """Determines pseudocount offset to shrink
+    LFCs asssociated with non-expressed genes to zero.
+
+    Parameters
+    ----------
+    scales_a
+        Scales in first population
+    scales_b
+        Scales in second population
+    where_zero_a
+        mask where no observed counts
+    where_zero_b
+        mask where no observed counts
+    """
+
+    max_scales_a = np.max(scales_a, 0)
+    max_scales_b = np.max(scales_b, 0)
+    assert max_scales_a.shape == where_zero_a.shape
+    assert max_scales_b.shape == where_zero_b.shape
+    assert where_zero_a.shape == where_zero_b.shape
+
+    if where_zero_a.sum() >= 1:
+        artefact_scales_a = max_scales_a[where_zero_a]
+        eps_a = np.percentile(artefact_scales_a, q=90)
```

Comment:
maybe should make this percentile an argument with default value

Reply:
Do you think this should be a parameter in `get_bayes_factors`? I replaced this default value in `estimate_pseudocounts_offset` but not in get_bayes_factors yet